og.module

```diff
@@ -44,6 +45,62 @@ function og_entity_insert(EntityInterface $entity) {
 }
 
 /**
+ * Implements hook_og_group_created().
+ */
+function og_og_group_created($entity_type, $bundle) {
+  $fields = OgGroupAudienceHelper::getAllGroupAudienceFields('user', 'user');
+
+  foreach ($fields as $field) {
+
+    if ($field->getFieldStorageDefinition()->getSetting('target_type') == $entity_type) {
+
+      if (!$field->getSetting('handler_settings')['target_bundles']) {
+        return;
+      }
+
+      if (in_array($bundle, $field->getSetting('handler_settings')['target_bundles'])) {
+        return;
+      }
+    }
+  }
+
+  // If we reached here, it means we need to create a field.
+  // Pick an unused name.
```

Comment:
Yes, this was one of the first things we did when I started working on OG. This is essentially the base of OG in my mind :) We don't just need it for the edit form, but everywhere. Custom storage in the og_memberships table is just an implementation detail.